src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -962,38 +962,38 @@ class spell_warl_t4_2p_bonus : public AuraScript
     }
 };
 
-// 30108, 34438, 34439, 35183 - Unstable Affliction
+// 316099 - Unstable Affliction
 class spell_warl_unstable_affliction : public AuraScript
 {
     bool Validate(SpellInfo const* /*spellInfo*/) override
     {
-        return ValidateSpellInfo({ SPELL_WARLOCK_UNSTABLE_AFFLICTION_DISPEL });
+        return ValidateSpellInfo({ SPELL_WARLOCK_UNSTABLE_AFFLICTION_DAMAGE, SPELL_WARLOCK_UNSTABLE_AFFLICTION_ENERGIZE });
     }
 
     void HandleDispel(DispelInfo* dispelInfo)
     {
         if (Unit* caster = GetCaster())
         {
-            if (AuraEffect const* aurEff = GetEffect(EFFECT_1))
-            {
-                if (Unit* target = dispelInfo->GetDispeller()->ToUnit())
-                {
-                    int32 bp = aurEff->GetAmount();
-                    bp = target->SpellDamageBonusTaken(caster, aurEff->GetSpellInfo(), bp, DOT);
-                    bp *= 9;
-
-                    // backfire damage and silence
-                    CastSpellExtraArgs args(aurEff);
-                    args.AddSpellBP0(bp);
-                    caster->CastSpell(target, SPELL_WARLOCK_UNSTABLE_AFFLICTION_DISPEL, args);
-                }
-            }
+            float sp = caster->SpellBaseDamageBonusDone(GetSpellInfo()->GetSchoolMask()) * 0.30613f;
```

Comment:
Where does this value come from?

Reply:
Effect 1: Id 6 (SPELL_EFFECT_APPLY_AURA)
Difficulty: Id 0 (DIFFICULTY_NONE)
BasePoints = 0.00 + spellPower * 0.30613
Targets (6, 0) (TARGET_UNIT_TARGET_ENEMY, NO_TARGET)
Aura Id 3 (SPELL_AURA_PERIODIC_DAMAGE), value = 0, misc = 0 (0), miscB = 0, amplitude = 0, periodic = 2000

Reply:
Hardcoding the value was a bad idea (already changed for 11.0) - it is available as a field in SpellEffect

Reply:
oh didnt know that, thanks for the tip